RFunction.R

```diff
@@ -60,374 +40,374 @@ rFunction = function(data, travelcut,
+        "the workflow BEFORE this MoveApp to add the sought columns.")
+    )
+  } else {
+    logger.info(" |- Sunrise and sunset columns identified. Able to perform night-time identification.")
+  }
+  
+  
+  
+  logger.info(" |- Input is in correct format. Proceeding with data preparation.")
+  
+  
+  # Data Preparation ===========================================================
+  
+  logger.info("Initiate Data Preparation Steps")
+  
+  
+  ## Generate overarching variables  ------------------------------
+
+  logger.info(" |- Generate overarching variables")
   
-  # Generate necessary data:
   data %<>% dplyr::mutate(
     ID = mt_track_id(.),
     timestamp = mt_time(.)
   )
   
+  #' NOTE: `yearmonthday`, `hourmin`, `timediff_hrs`, `dist_m` & `kmph`
+  #' variables are expected to provide information between consecutive
+  #' locations. If the Standardizing App (or other) has been used earlier in the
+  #' WF, these cols could already be present in the input. However, there is no
+  #' guarantee that input has been thinned by other in-between App. Therefore, to
+  #' ensure accuracy, we always (re)generate these columns here.
+  
+  # Add date label and day-hours (i.e. decimal hours since start of day)
+  # QUESTION (BC): Should we restrict usage to input with local_time only? I think we should
   if ("timestamp_local" %in% colnames(data)) {
-    data %<>% mutate(yearmonthday = stringr::str_replace_all(stringr::str_sub(timestamp_local, 1, 10), "-", ""))
+    data %<>% 
+      mutate(
+        yearmonthday = stringr::str_replace_all(stringr::str_sub(timestamp_local, 1, 10), "-", ""),
+        hourmin = lubridate::hour(timestamp_local) + 
+          lubridate::minute(timestamp_local)/60 + 
+          lubridate::second(timestamp_local)/3600
+      ) 
   } else {
-    data %<>% mutate(yearmonthday = stringr::str_replace_all(stringr::str_sub(timestamp, 1, 10), "-", ""))                     
+    data %<>% 
+      mutate(
+        yearmonthday = stringr::str_replace_all(stringr::str_sub(timestamp, 1, 10), "-", ""),
+        hourmin = lubridate::hour(timestamp) + 
+          lubridate::minute(timestamp)/60 + 
+          lubridate::second(timestamp)/3600
+      )                     
   }
   
-  # Generate necessary data:
-  data$dist_m <- as.vector(move2::mt_distance(data))
-  data$kmph <- move2::mt_speed(data) %>% units::set_units("km/h") %>% as.vector()
-  #data$timediff_hrs <- lubridate::mt_time_lags(data)
+  # Add time gap, distance and speed cols
   data %<>% 
     arrange(mt_track_id(data), mt_time(data)) %>%
     # distinct(timestamp, .keep_all = TRUE) %>%
     mutate(
-      #dist_m = sqrt((x - lag(x, default = NA))^2 + (y - lag(y, default = NA))^2),
-      timediff_hrs =   mt_time_lags(.) %>%
-        units::set_units("hours") %>%
-        as.vector(),
-      #kmph = move2::mt_speed(data),
-      hour = lubridate::hour(mt_time(data))
+      timediff_hrs = as.vector(mt_time_lags(., units = "hours")),
```

Comment:
question: is this computed on the basis of the tag time or the local time?  I ask because i forgot to eliminate high speed points in the standardising app just now and ended up with time differences being calculated between two different time zones.  not sure if it ultimately makes a difference but we might want to make sure....

Reply:
`mt_time_lags()` uses the "time column" defined in the input move2 object, which by default in Movebank is column `timestamp`, i.e. tag time in UTC. 

But yeah, that function could generate inaccurate lags if local time is set as the "time column", and if the track crosses different timezones